### DISCUSSION TITLE
Override type resolution for a single type

### DISCUSSION THREAD

**Discussion by rrva** [graphql-java-kickstart/graphql-java-tools#718] [Q&A]
If you use the automatic wiring for types, is there a way to override the wiring/type resolution for a single type but keep the rest?

I have a GraphQL interface `Fruit` which has subtypes `Apple` and `Banana`. There exists a corresponding java interface `Fruit` and java classes `Apple` and `Banana` . 

How do I introduce a new class only on the java side, lets say called `Orange`, which on the java side implements `Fruit`, but I want it to resolve to `Banana` on the GraphQL side (so I do not have to introduce another GraphQL type in the schema).

**Comment by rrva** [comment]
This gets me half-way:

```kotlin

    val executableSchema = schemaParser.makeExecutableSchema()
    val oldFruitTypeResolver =
        executableSchema.codeRegistry.getTypeResolver(GraphQLInterfaceType.newInterface().name("Fruit").build())

    val fruitTypeResolver = TypeResolver { env ->
        when {
            env.getObject<Any>() is Orange -> env.schema.getObjectType("Banana")
            else -> oldFruitTypeResolver.getType(env)
        }
    }

    val schemaObjects = schemaParser.parseSchemaObjects()

    return GraphQLSchema.newSchema()
        .description(schemaObjects.description)
        .query(schemaObjects.query)
        .mutation(schemaObjects.mutation)
        .subscription(schemaObjects.subscription)
        .additionalTypes(schemaObjects.dictionary)
        .additionalDirectives(schemaObjects.directives)
        .codeRegistry(

            schemaObjects.codeRegistryBuilder
                .typeResolver("Fruit", fruitTypeResolver)
                .build()
        )
        .build()
```

but then I end up building a schema which does not have the correct data fetchers

**Comment by rrva** [comment]
@oryan-block do you if there is an easy solution with the current graphql-java-tools for this? I am even considering departing graphql-java-tools and using pure graphql-java for this type of issue, but this is substantial work on my end and I loose out on all of the niceties of auto-wiring.

**Reply by oryan-block** [reply]
Sorry I didn't have time to look at this sooner.
Did you try using a `TypeDefinitionFactory`? This can be done to add GraphQL types dynamically even if they are not present in the schema.

https://www.graphql-java-kickstart.com/tools/type-definition-factory/

**Reply by rrva** [reply]
Thanks I forgot to mark this one as resolved. https://github.com/graphql-java/graphql-java/discussions/3013#discussioncomment-4127287
